openedx/core/djangoapps/user_api/tests/test_views.py

```diff
@@ -1405,6 +1468,56 @@ def test_register_with_profile_info(self):
         self.assertEqual(account_settings["goals"], self.GOALS)
         self.assertEqual(account_settings["country"], self.COUNTRY)
 
+    @override_settings(REGISTRATION_EXTENSION_FORM='openedx.core.djangoapps.user_api.tests.test_helpers.TestCaseForm')
+    @mock.patch('openedx.core.djangoapps.user_api.tests.test_helpers.TestCaseForm.DUMMY_STORAGE')
+    def test_with_extended_form(self, storage_dict):
+        test_dict = {}
+
+        def getter(name):
+            """
+            Get the value from the test dictionary.
+            """
+            return test_dict[name]
+
+        def setter(name, val):
+            """
+            Set the value in the test dictionary.
+            """
+            test_dict[name] = val
+
+        storage_dict.__getitem__.side_effect = getter
+        storage_dict.__setitem__.side_effect = setter
```

Comment:
I don't suppose there's a way to replace the getter/setter above with just `@mock.patch('openedx....DUMMY_STORAGE', new_callable=dict)` ?